common/hello.go

```diff
@@ -0,0 +1,190 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/logger"
+	"github.com/giantswarm/clustertest/pkg/organization"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/cluster-test-suites/internal/state"
+)
+
+func runHelloWorld(externalDnsSupported bool) {
+	Context("hello world", func() {
+		var nginxApp, helloApp *v1alpha1.App
+		var nginxConfigMap, helloConfigMap *v1.ConfigMap
+		var helloWorldIngressHost string
+
+		BeforeEach(func() {
+			if !externalDnsSupported {
+				Skip("Bastion is not supported.")
+			}
+
+			ctx := context.Background()
+			org := state.GetCluster().Organization
+
+			// The hello-world app ingress requires a `Certificate` and a DNS record, so we need to make sure `cert-manager` and `external-dns` are deployed.
+			Eventually(func() error {
+				app, err := state.GetFramework().GetApp(ctx, fmt.Sprintf("%s-cert-manager", state.GetCluster().Name), org.GetNamespace())
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(3 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+
+			Eventually(func() error {
+				app, err := state.GetFramework().GetApp(ctx, fmt.Sprintf("%s-external-dns", state.GetCluster().Name), org.GetNamespace())
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(3 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+
+			// `external-dns` will only create dns records for Ingresses on the `kube-system` namespace, that's why we install the nginx app in that namespace.
```

Comment:
[minor]

Can we be specific here, since this may change in the future? I think this refers to the default value `namespaceFilter: kube-system` in external-dns-app?